enterprise/models.py

```diff
@@ -908,6 +908,57 @@ def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
 
 
 @python_2_unicode_compatible
+class EnterpriseCustomerCatalog(TimeStampedModel):
+    """
+    Store catalog information from course discovery specifically for Enterprises.
+
+    We use this model to consolidate course catalog information, which includes
+    information about catalogs, courses, programs, and possibly more in the
+    future, as the course discovery service evolves.
+    """
+
+    uuid = models.UUIDField(
+        primary_key=True,
+        default=uuid4,
+        editable=False
+    )
+    enterprise_customer = models.ForeignKey(
+        EnterpriseCustomer,
+        blank=False,
+        null=False,
+        related_name='enterprise_customer_catalog',
+        on_delete=models.deletion.CASCADE
+    )
+    query = models.TextField(
+        help_text=_('Query to the course discovery service. Leave empty for all results.')
+    )
+    history = HistoricalRecords()
+
+    class Meta(object):
+        verbose_name = _("Enterprise Catalog")
+        verbose_name_plural = _("Enterprise Catalogs")
```

Comment:
Woops! Not sure how this escaped naming changes :s I must've forgotten to remake migrations.